### PR TITLE
create ProofOfGoodEntry endpoint

### PR DESCRIPTION
**Pull request by elijaholmos**
Implemented the following endpoints:
- create Proof of Good entry
- retrieve PoG entry by ID
- retrieve all PoG entries

no implementation to update a PoG entry because of blockchain mutability concerns. if that is dismissible, I will add a PUT endpoint.

### CODE REVIEW

**Comment by elijaholmos** [comment]
removed `src\controllers\open-api.controller.ts` because of a jsonschema error
image

File: src/repositories/index.ts [hunk 1]
```diff
@@ -1 +1,4 @@
+export * from './good-activity.repository';
 export * from './good-oracle.repository';
+export * from './good-category.repository';
+export * from './proof-of-good-entry.repository';
```
Comment (kharaone) [inline src/repositories/index.ts]: small comment on naming convention. Should we rename `proof-of-good-entry` into `good-entry`

File: src/models/index.ts [hunk 2]
```diff
@@ -6,3 +6,4 @@ export * from './good-category.model';
 export * from './good-oracle-input.model';
 export * from './good-oracle.model';
 export * from './proof-of-good-entry-input.model';
+export * from './proof-of-good-entry.model';
```
Comment (kharaone) [inline src/models/index.ts]: naming convention consistency, should we rename these into `good-entry` and `good-entry-input` ?

File: src/models/good-category.model.ts [hunk 3]
```diff
@@ -15,45 +16,51 @@ export class GoodCategory {
...
    *
    */
-  @property({required: true, jsonSchema: {
-  type: 'string',
-}})
-  id: string;
+  @property({
+    id: true,
+    jsonSchema: {
+      type: 'string',
```
Comment (kharaone) [inline src/models/good-category.model.ts]: Probably better to have it decorated as an integer

```suggestion
      type: 'number',
```

File: src/controllers/category.controller.ts [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+          },
+        },
+      },
+      description: '',
+    })
+    category: GoodCategoryInput | any,
+  ): Promise<unknown> {
+    delete category?.id;
+    return this.repository.updateById(id, category).catch((err: Error) => {
+      if (err.message == '404')
```
Comment (kharaone) [inline src/controllers/category.controller.ts]:
```suggestion
      if (err.message == 'Document not found')
```
to stay consistent with the error handling of BetterFirestoreConnector

File: src/controllers/activity.controller.ts [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+            },
+          },
+        },
+      },
+    })
+    activity: GoodActivityInput | any,
+  ): Promise<unknown> {
+    delete activity?.id;
+    return this.repository.updateById(id, activity).catch((err: Error) => {
+      if (err.message == '404')
```
Comment (kharaone) [inline src/controllers/activity.controller.ts]:
```suggestion
      if (err.message == 'Document not found')
```

Comment (kharaone) [reply src/controllers/activity.controller.ts]: which means L169 of src\connectors\BetterFirestoreConnector.ts needs to be changed

**Review by kharaone — CHANGES_REQUESTED** [review]
Looks good as a whole.

I have a few changes requests in order to align with the excel document AND be consistent

File: src/models/index.ts (hunk 2 above)
Comment (elijaholmos) [reply src/models/index.ts]: no objection from me

File: src/repositories/index.ts (hunk 1 above)
Comment (elijaholmos) [reply src/repositories/index.ts]: resolved w 90d258a2744c1efa285c8932754c5b5b2203cee9

File: src/models/index.ts (hunk 2 above)
Comment (elijaholmos) [reply src/models/index.ts]: resolved w 90d258a2744c1efa285c8932754c5b5b2203cee9

File: src/controllers/category.controller.ts [hunk 6]
```diff
@@ -213,7 +213,7 @@ export class CategoryController {
   ): Promise<unknown> {
     delete category?.id;
     return this.repository.updateById(id, category).catch((err: Error) => {
-      if (err.message == '404')
+      if (err.message == 'Document not found')
         throw new HttpErrors.NotFound('Oracle Not Found');
```
Comment (kharaone) [inline src/controllers/category.controller.ts]:
```suggestion
        throw new HttpErrors.NotFound('Category Not Found');
```

**Review by kharaone — CHANGES_REQUESTED** [review]
Looks Good To Me

Just an oversight on the error message to be corrected, and we're good :)

**Pull request merged**